src/action_install.rs

```diff
@@ -1,77 +1,82 @@
-use crate::aur_rpc_utils;
 use crate::pacman;
 use crate::reviewing;
 use crate::rua_files;
 use crate::tar_check;
 use crate::terminal_util;
 use crate::wrapped;
 
+use aur_depends::{Actions, Missing, Resolver};
 use directories::ProjectDirs;
 use fs_extra::dir::CopyOptions;
-use indexmap::IndexMap;
-use indexmap::IndexSet;
 use itertools::Itertools;
 use log::debug;
 use log::trace;
-use raur::Package;
-use std::collections::HashSet;
 use std::fs;
 use std::fs::ReadDir;
 use std::path::PathBuf;
 
-pub fn install(targets: &[String], dirs: &ProjectDirs, is_offline: bool, asdeps: bool) {
-	let alpm = pacman::create_alpm();
-	let (split_to_raur, pacman_deps, split_to_depth) =
-		aur_rpc_utils::recursive_info(targets, &alpm).unwrap_or_else(|err| {
-			panic!("Failed to fetch info from AUR, {}", err);
-		});
-	let split_to_pkgbase: IndexMap<String, String> = split_to_raur
-		.iter()
-		.map(|(split, raur)| (split.to_string(), raur.package_base.to_string()))
-		.collect();
-	let not_found = split_to_depth
-		.keys()
-		.filter(|pkg| !split_to_raur.contains_key(*pkg))
-		.collect_vec();
-	if !not_found.is_empty() {
-		eprintln!(
-			"Need to install packages: {:?}, but they are not found on AUR.",
-			not_found
-		);
+pub fn install(
+	resolver: Resolver,
+	targets: &[String],
+	dirs: &ProjectDirs,
+	is_offline: bool,
+	as_deps: bool,
+) {
+	let actions = resolver
+		.resolve_targets(targets)
+		.unwrap_or_else(|e| panic!("failed to resolver deps {}", e));
+
+	if !actions.missing.is_empty() {
+		eprintln!("Need to install packages: but they are not found:");
+
+		for Missing { stack, dep } in actions.missing {
+			eprint!("{}", dep);
+			if !stack.is_empty() {
+				eprint!(" (Wanted by: {})", stack.join(" -> "));
```

Comment:
One other possible improvement that I had in mind is to let the user see the whole dependency tree during "installation overview" (this function here).
Potential nice things about it: show all unresolved things at once, is useful even if you have no errors, and you can meaningfully highlight different things with color (like dependency recursion depth).